lib/combineNodes.js

```diff
@@ -0,0 +1,290 @@
+        var accessorId = attributes[attributeId];
+        var attributeAccessor = accessors[accessorId];
+        checkCombineWarning(attributeAccessor, 'accessor', accessorId);
+
+        var attributeBufferView = bufferViews[attributeAccessor.bufferView];
+        checkCombineWarning(attributeBufferView, 'bufferView', attributeAccessor.bufferView);
+
+        var attributeBuffer = buffers[attributeBufferView.buffer];
+        checkCombineWarning(attributeBuffer, 'buffer', attributeBufferView.buffer);
+
+        var bufferViewSource = attributeBuffer.extras._pipeline.source.slice(attributeBufferView.byteOffset, attributeBufferView.byteOffset + attributeBufferView.byteLength);
+        var accessorSource = bufferViewSource.slice(attributeAccessor.byteOffset, attributeAccessor.byteOffset + attributeAccessor.count * defaultValue(attributeAccessor.byteStride, 12));
+        var premultipliedSource = premultiplyTransformWithBuffer(transform, accessorSource, attributeId, attributeAccessor.min, attributeAccessor.max);
+        var premultipliedSourceLength = premultipliedSource.length;
+
+        var newAccessorId = createNewId(gltf, meshId, attributeId, 'accessor');
+        var newBufferId = createNewId(gltf, meshId, attributeId, 'buffer');
+        var newBufferViewId = createNewId(gltf, meshId, attributeId, 'bufferView');
+
+        var newBuffer = {
+            "type": "arraybuffer",
+            "byteLength": premultipliedSourceLength,
+            "uri": "data:,",
+            "extras": {
+                "_pipeline": {
+                    "extension": ".bin",
+                    "source": premultipliedSource,
+                    "deleteExtras": true
+                }
+            }
+        };
+        var newBufferView = {
+            "buffer": newBufferId,
+            "byteOffset": 0,
+            "byteLength": premultipliedSourceLength,
+            "extras": {
+                "_pipeline": {
+                    "deleteExtras": true
+                }
+            }
+        };
+
+        if (defined(attributeBufferView.target)) {
+            newBufferView.target = attributeBufferView.target;
+        }
+
+        var newAccessor = clone(attributeAccessor, true);
+        newAccessor.byteOffset = 0;
+        newAccessor.bufferView = newBufferViewId;
+
+        attributes[attributeId] = newAccessorId;
+        accessors[newAccessorId] = newAccessor;
+        bufferViews[newBufferViewId] = newBufferView;
+        buffers[newBufferId] = newBuffer;
+    }
+}
+
+//Returns a new buffer with the elements premultiplied by the transform matrix.
+function premultiplyTransformWithBuffer(matrix, buffer, attributeId, min, max) {
+    var bufferLength = buffer.length;
+    var resultBuffer = new Buffer(bufferLength);
+    var bufferIndex = 0;
+
+    while (bufferIndex < bufferLength) {
+        var currentVector = new Cartesian3(buffer.readFloatLE(bufferIndex), buffer.readFloatLE(bufferIndex + 4), buffer.readFloatLE(bufferIndex + 8));

```

Comment:
Use scratch variables here and in any tight loops.  This improves performance by reducing GC pressure.  See https://github.com/AnalyticalGraphicsInc/cesium/tree/master/Documentation/Contributors/CodingGuide#result-parameters-and-scratch-variables